include/alpaka/vec/Vec.hpp

```diff
@@ -927,6 +1000,83 @@ namespace alpaka
                                 vec);
                 }
             };
+
+            //#############################################################################
+            //! (Non-)Reverse specialization for 1D Vec.
+            //#############################################################################
+            template<
+                typename TSize>
+            struct Reverse<
+                Vec<dim::DimInt<1u>, TSize>>
+            {
+                ALPAKA_NO_HOST_ACC_WARNING
+                ALPAKA_FN_HOST_ACC static auto reverse(
+                    Vec<dim::DimInt<1u>, TSize> const & vec)
+                -> Vec<dim::DimInt<1u>, TSize>
+                {
+                    return vec;
+                }
+            };
+        }
+
+        namespace detail
+        {
+            //#############################################################################
+            //! A function object that returns the value at the index from the back of the vector.
+            //#############################################################################
+            template<
+                std::size_t Tidx>
+            struct CreateConcat
+            {
+                //-----------------------------------------------------------------------------
+                //!
+                //-----------------------------------------------------------------------------
+                ALPAKA_NO_HOST_ACC_WARNING
+                template<
+                    typename TDimL,
+                    typename TDimR,
+                    typename TSize>
+                ALPAKA_FN_HOST_ACC static auto create(
+                    Vec<TDimL, TSize> const & vecL,
+                    Vec<TDimR, TSize> const & vecR)
+                -> TSize
+                {
+                    return Tidx < TDimL::value ? vecL[Tidx] : vecR[Tidx - TDimL::value];
+                }
+            };
+        }
+        namespace traits
+        {
+            //#############################################################################
+            //! Concatenation specialization for Vec.
+            //#############################################################################
+            template<
+                typename TDimL,
+                typename TDimR,
+                typename TSize>
+            struct Concat<
+                Vec<TDimL, TSize>,
+                Vec<TDimR, TSize>>
+            {
+                ALPAKA_NO_HOST_ACC_WARNING
+                ALPAKA_FN_HOST_ACC static auto concat(
+                    Vec<TDimL, TSize> const & vecL,
+                    Vec<TDimR, TSize> const & vecR)
+                -> Vec<dim::DimInt<TDimL::value + TDimR::value>, TSize>
+                {
+                    return
+#ifdef ALPAKA_CREATE_VEC_IN_CLASS
```

Comment:
only a note: we should check if we can skip this `#ifdef` asap, it is very hard to read this little peace of code.